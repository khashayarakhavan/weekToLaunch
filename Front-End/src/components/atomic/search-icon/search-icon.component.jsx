import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import sprite from "../../../assets/spriteSVG/sprite-medals.svg";
import CrownIcon from "../../../assets/SVG/crown.svg";
import { toggleCartHidden } from '../../../redux/cart/cart.actions';
import { selectCartItemsCount } from '../../../redux/cart/cart.selectors';
import {
  SearchContainer,
  StarIcon,
  SVG
} from './search-icon.styles';
import "./search-icon.styles.scss";
/* 
* how to import SVG in React.js
*/
export const SearchBox = ({ toggleCartHidden, itemCount }) => (
         <SearchContainer>
           {/* -- Style 1 how to import SVG: direct SVG file & edit partly using Sass -- */}
           <CrownIcon className="nanai-icon" onClick={toggleCartHidden} />

           {/* -- Style 2 to import SVG: SVG file inside StyledComponents -- */}
           <StarIcon />

           {/* -- Style 3 to import SVG: using Sprite bundle + StyledComponents "Recommended Method" -- */}
           <SVG onClick={toggleCartHidden}>
             <use href={sprite + "#icon-trophy"} />
           </SVG>
         </SearchContainer>
       );

const mapDispatchToProps = dispatch => ({
  toggleCartHidden: () => dispatch(toggleCartHidden())
});

const mapStateToProps = createStructuredSelector({
  itemCount: selectCartItemsCount
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SearchBox);
