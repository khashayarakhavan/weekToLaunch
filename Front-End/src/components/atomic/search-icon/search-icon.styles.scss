.cart-icon {
  width: 100px;
  height: 100px;
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;

  .nanai-icon {
    width: 10px;
    height: 10px;
    fill: blue;
  }
  .shopping-icon {
    width: 24px;
    height: 24px;
    fill: orange;
  }
  .trophy-icon {
    width: 24px;
    height: 24px;
    fill: orangered;
    
  }

  .item-count {
    position: absolute;
    font-size: 10px;
    font-weight: bold;
    bottom: 12px;
  }
}

.nanai-icon {
    width: 30px;
    height: 30px;
    fill: yellow;
  }