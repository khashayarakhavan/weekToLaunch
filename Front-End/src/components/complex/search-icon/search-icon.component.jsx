import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import sprite from "../../assets/spriteSVG/sprite-medals.svg";

import { toggleCartHidden } from '../../redux/cart/cart.actions';
import { selectCartItemsCount } from '../../redux/cart/cart.selectors';

import { ReactComponent as CrownIcon } from '../../assets/SVG/crown.svg';

import "./search-icon.styles.scss";
import {
  SearchContainer,
  StarIcon,
  SVG
} from './search-icon.styles';

export const SearchBox = ({ toggleCartHidden, itemCount }) => (
         <SearchContainer>
           {/* -- Style 1 to import SVG: direct SVG file & edit partly using Sass -- */}
           <CrownIcon className="nanai-icon" onClick={toggleCartHidden} />

           {/* -- Style 2 to import SVG: SVG file inside StyledComponents -- */}
           <StarIcon />

           {/* -- Style 3 to import SVG: using Sprite bundle + StyledComponents "Recommended Method" -- */}
           <SVG onClick={toggleCartHidden}>
             <use href={sprite + "#icon-trophy"} />
           </SVG>
         </SearchContainer>
       );

const mapDispatchToProps = dispatch => ({
  toggleCartHidden: () => dispatch(toggleCartHidden())
});

const mapStateToProps = createStructuredSelector({
  itemCount: selectCartItemsCount
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SearchBox);
