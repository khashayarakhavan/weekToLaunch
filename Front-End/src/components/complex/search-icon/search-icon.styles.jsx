import styled from 'styled-components';

import {ReactComponent as StarSvg} from "../../assets/SVG/star.svg";

export const SearchContainer = styled.div`
  width: 100px;
  height: 100px;
  position: relative;
  display: flex;
  align-items: center;
  justify-content: space-around;
  
`;

export const SVG = styled.svg`
  width: 24px;
  height: 24px;
  fill: lightblue;
 
  cursor: pointer;
`;

SearchContainer.displayName = 'SearchContainer';

export const StarIcon = styled(StarSvg)`
  width: 24px;
  height: 24px;
  fill: pink;

`;

export const ItemCountContainer = styled.span`
  position: absolute;
  font-size: 10px;
  font-weight: bold;
  bottom: 12px;
`;

ItemCountContainer.displayName = 'ItemCountContainer';
