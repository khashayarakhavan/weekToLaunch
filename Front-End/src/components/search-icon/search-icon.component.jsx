import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { toggleCartHidden } from '../../redux/cart/cart.actions';
import { selectCartItemsCount } from '../../redux/cart/cart.selectors';
import {
  SearchContainer,
  StarIcon
} from './search-icon.styles';
import "./search-icon.styles.scss";

export const SearchBox = ({ toggleCartHidden, itemCount }) => (
         <SearchContainer>
           {/* -- Style 1 to import SVG: direct SVG file & edit partly using Sass -- */}
           {/* <CrownIcon id="nanai-icon" className="nanai-icon" onClick={toggleCartHidden} /> */}

           {/* -- Style 2 to import SVG: SVG file inside StyledComponents -- */}
           <StarIcon className="star" />

           {/* -- Style 3 to import SVG: using Sprite bundle + StyledComponents "Recommended Method" -- */}
           {/* <SVG onClick={toggleCartHidden}>
             <use href={sprite + "#icon-trophy"} />
           </SVG> */}
         </SearchContainer>
       );

const mapDispatchToProps = dispatch => ({
  toggleCartHidden: () => dispatch(toggleCartHidden())
});

const mapStateToProps = createStructuredSelector({
  itemCount: selectCartItemsCount
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SearchBox);
