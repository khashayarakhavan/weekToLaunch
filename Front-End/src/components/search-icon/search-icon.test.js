import { shallow } from 'enzyme';
import React from 'react';
import { SearchBox } from "./search-icon.component";

describe("SearchBox component", () => {
  let wrapper;
  let mockToggleCartHidden;
  beforeEach(() => {
    mockToggleCartHidden = jest.fn();

    const mockProps = {
      itemCount: 0,
      toggleCartHidden: mockToggleCartHidden,
    };

    wrapper = shallow(<SearchBox {...mockProps} />);
  });

  it("should render CartIcon component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call toggleCartHidden when icon is clicked", () => {
    wrapper.find('#nanai-icon').simulate("click");
    expect(mockToggleCartHidden).toHaveBeenCalled();
  });

});
