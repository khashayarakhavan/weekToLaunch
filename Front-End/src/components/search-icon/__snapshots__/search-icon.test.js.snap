// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox component should render CartIcon component 1`] = `ShallowWrapper {}`;
